wgpu-core/src/command/render.rs

```diff
@@ -605,6 +614,41 @@ impl<'a, A: HalApi> RenderPassInfo<'a, A> {
         let mut extent = None;
         let mut sample_count = 0;
 
+        let mut detected_mutiview: Option<Option<NonZeroU32>> = None;
+
+        let mut check_multiview = |view: &TextureView<A>| {
+            // Get the multiview configuration for this texture view
+            let layers = view.selector.layers.end - view.selector.layers.start;
+            let this_multiview = if layers == 1 {
+                None
+            } else {
+                let layers = NonZeroU32::new(layers).expect("Non-zero layers expected");
+                Some(layers)
+            };
+
+            // Make sure that if this view is a multiview, it is set to be an array
+            if this_multiview.is_some() && view.desc.dimension != TextureViewDimension::D2Array {
+                return Err(RenderPassErrorInner::MultiViewDimensionMismatch);
+            }
+
+            // Validate matching first, or store the first one
+            if let Some(multiview) = detected_mutiview {
+                if multiview != this_multiview {
+                    return Err(RenderPassErrorInner::MultiViewMismatch);
+                }
+            } else {
+                // Multiview is only supported if the feature is enabled
+                if this_multiview.is_some() && !device.features.contains(wgt::Features::MULTIVIEW) {
+                    return Err(RenderPassErrorInner::MissingFeatures(MissingFeatures(
+                        wgt::Features::MULTIVIEW,
+                    )));
+                }
+
+                detected_mutiview = Some(this_multiview);
```

Comment:
I think this logic isn't going to work right. It scans several attachments and only errors if we are switching from multiview to non-multiview. Imagine if the first analyzed attachment is non-layered but the second is layered. We should error, but this could wouldn't.

Reply:
The implemented logic here on every view is:
- check if multiview on or off has already been detected
- if it has: check if the multiview on/off and layer count matches
- if it hasn't: use the current view's value as the multiview detection

Subsequent checks will be checked against the first one's detected value. This is what the state of `detected_multiview` means:
- None: Not detected
- Some(None): Detected, no multiview
- Some(Some(2)): Detected, multiview with two layers